staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -76,38 +77,85 @@ const (
 	zonesKey = "zones"
 )
 
-// replaceTopology overwrites an existing topology key by a new one.
+// replaceTopology overwrites an existing key by a new one.
+// If there are any newKey already exist in a term, we will combine the
+// replaced key Values with the existing ones
 func replaceTopology(pv *v1.PersistentVolume, oldKey, newKey string) error {
+	// Make sure the necessary fields exist
+	if pv == nil || pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil ||
+		pv.Spec.NodeAffinity.Required.NodeSelectorTerms == nil || len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) == 0 {
+		return nil
+	}
 	for i := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms {
+		newKeyValsMap := make(map[string]bool)
+		var requirementsToRemove []int
 		for j, r := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions {
 			if r.Key == oldKey {
 				pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions[j].Key = newKey
 			}
+			if r.Key == newKey || r.Key == oldKey {
+				for _, v := range r.Values {
+					newKeyValsMap[v] = true
+				}
+				requirementsToRemove = append(requirementsToRemove, j)
+			}
+		}
+
+		// Combine any duplicated keys
+		if len(requirementsToRemove) > 1 {
+			// This means there are duplicated new Labels, we shall combine them together
+			for j := len(requirementsToRemove) - 1; j >= 0; j-- {
+				index := requirementsToRemove[j]
+				ncRequirements := pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions
+				ncRequirements[len(ncRequirements)-1], ncRequirements[index] = ncRequirements[index], ncRequirements[len(ncRequirements)-1]
+				pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions = ncRequirements[:len(ncRequirements)-1]
+			}
+			// after remove all, append the new value
+			var newKeyVals []string
+			for k := range newKeyValsMap {
```

Comment:
nit: `k` -> `val`, since `k` could be confused with topology key

Reply:
Done